lb_film_and_user_recommender.py

```diff
@@ -0,0 +1,255 @@
+    :return: a dictionary where keys are users and values are the user's score based on number of liked films etc.
+    """
+    def similarity_scorer(myscore, theirscore, avgrating, ratingcount):
+        """
+        Arbitrary algorithm to give a score based on our original user's rating, the similar user's rating, their
+        difference from the average rating, and the obscurity of the film based on its rating count
+
+        :param myscore: (float) original user's score
+        :param theirscore: (float) similar user's score
+        :param avgrating: (float) the film's average rating
+        :param ratingcount: (int) the number of ratings the film has overall
+        :return: (float) returns a float score
+        """
+        # return (((6 * myscore + 4 * theirscore) - 2 * avgrating) ** 2) * 100 / math.sqrt(ratingcount)
+        return ((4 * myscore + 6 * theirscore) - 10 * avgrating) * (100 / math.sqrt(ratingcount))
+
+    str_score_dict = {
+        0.0: '0',
+        0.5: '0.5',
+        1.0: '1',
+        1.5: '1.5',
+        2.0: '2',
+        2.5: '2.5',
+        3.0: '3',
+        3.5: '3.5',
+        4.0: '4',
+        4.5: '4.5',
+        5.0: '5'
+    }
+
+    totalfilms = len(films)
+    for film in films:
+        webpage_number = 0
+        main_page = urllib.request.urlopen('https://letterboxd.com/film/' + film + '/')
+        main_page = str(list(main_page))
+        clean_title = re.findall('data-film-name="(.+?)"', main_page)[0]
+        try:
+            avgrating = float(re.findall('"ratingValue":(.+?),"description"', main_page)[0])
+        except:
+            avgrating = 3
+        try:
+            ratingcount = int(re.findall('"ratingCount":(.+?),"worstRating"', main_page)[0])
+        except:
+            ratingcount = 30
+
+        print('Checking high ratings for "%s"...' % clean_title)
+
+        while True:
+            webpage_number += 1
+            print('Checking page %d...' % webpage_number)
+            ratings_page = str(list(urllib.request.urlopen('https://letterboxd.com/film/%s/ratings/page/%d/' % (film, webpage_number))))
+            found_something = False
+            x = 5.0
+            while x >= lowerbound:
+                ratings_block = re.findall('<h2><span class="rating rating-large rated-large-' + str(int(x*2)) + '">(.+?)</ul>', ratings_page)
+                if ratings_block:
+                    found_something = True
+                    for user in re.findall('href="/(.+?)/"', ratings_block[0]):
+                        scoretoadd = similarity_scorer(myscore, x, avgrating, ratingcount)
+                        similar_user_dict[user] = similar_user_dict.get(user, 0) + scoretoadd
+                x -= 0.5
+            if (found_something == False) and (x < lowerbound):
+                totalfilms -= 1
+                print('"%s" done! %d %s/5 film(s) remaining. \n' % (clean_title, totalfilms, str_score_dict[myscore]))
+                break
+    return similar_user_dict
```

Comment:
Generally a good idea to avoid modifying the objects you pass to your function, better control of side-effects. I can see why you'd do that here though, but might be an idea to change the function name and not return anything to avoid mistakes.